querydsl-core/src/main/java/com/querydsl/core/types/dsl/Expressions.java

```diff
@@ -1463,4 +1463,84 @@ private static Template createTemplate(String template) {
         return TemplateFactory.DEFAULT.create(template);
     }
 
+    /**
+     * Create a new BooleanExpression
+     *
+     * @param expr Expression of type Boolean
+     * @return new BooleanExpression
+     */
+    public static BooleanExpression asBoolean(Expression<Boolean> expr) {
+
+        return new BooleanExpression(expr) {
```

Comment:
I'd say extract the underlying mixin of the incoming `expr` by using `Expression<Boolean> underlyingMixin = ExpressionUtils.extract(expr)`.

then `return new BooleanExpression(underlyingMixin) {...}`.

The same for the rest also?
You never know what assumptions code elsewhere makes.